modules/programs/qcal.nix

```
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.programs.qcal;
+
+  qcalAccounts = lib.attrValues
+    (lib.filterAttrs (_: a: a.qcal.enable) config.accounts.calendar.accounts);
+
+  rename = oldname:
+    builtins.getAttr oldname {
+      url = "Url";
+      userName = "Username";
+      passwordCommand = "PasswordCmd";
+    };
+
+  filteredAccounts = lib.trivial.pipe qcalAccounts (map map [
+    # We're only interested in config options in the remote submodule
+    (builtins.getAttr "remote")
+    # Get only the attributes we care about
+    (lib.getAttrs [ "url" "userName" "passwordCommand" ])
+    (lib.filterAttrs (_: attr: !(isNull attr)))
+    # qcal expects these values under other names, so we rename the keys in the attribute set
+    (lib.mapAttrs' (name: value: lib.nameValuePair (rename name) value))
+    # qcal wants the password command as a string, not as a list of strings
+    (lib.mapAttrs (_: value:
+      if lib.isString value then value else lib.concatStringsSep " " value))
+  ]);
+
+in {
+  options.programs.qcal = {
+    enable = lib.mkEnableOption "qcal, a CLI calendar application";
+    timezone = lib.mkOption {
```

Please separate options by empty lines. E.g.,

```suggestion

    timezone = lib.mkOption {
```